gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java

```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
+    assertEquals(diskName, image.<DiskImageConfiguration>configuration().sourceDisk().disk());
+    assertNull(image.status());
+    assertNull(image.diskSizeGb());
+    assertNull(image.licenses());
+    assertNull(image.deprecationStatus());
+    // test get image
+    image = compute.get(imageId);
+    assertNotNull(image.id());
+    assertNotNull(image.imageId());
+    assertNotNull(image.creationTimestamp());
+    assertNotNull(image.configuration());
+    assertEquals(ImageConfiguration.Type.DISK, image.configuration().type());
+    assertTrue(image.configuration() instanceof DiskImageConfiguration);
+    assertEquals(diskName, image.<DiskImageConfiguration>configuration().sourceDisk().disk());
+    assertEquals(100L, (long) image.diskSizeGb());
+    assertNotNull(image.status());
+    assertNull(image.deprecationStatus());
+    // test deprecate image
+    DeprecationStatus<ImageId> deprecationStatus =
+        DeprecationStatus.builder(DeprecationStatus.Status.DEPRECATED, imageId)
+            .deprecated(System.currentTimeMillis())
+            .build();
+    operation = image.deprecate(deprecationStatus);
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    image = compute.get(imageId);
+    assertEquals(deprecationStatus, image.deprecationStatus());
+    remoteDisk.delete();
+    operation = image.delete();
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    assertNull(compute.get(imageId));
+  }
+
+  @Test
+  public void testListImages() {
+    Page<Image> imagePage = compute.listImages(IMAGE_PROJECT);
+    Iterator<Image> imageIterator = imagePage.iterateAll();
+    while (imageIterator.hasNext()) {
+      Image image = imageIterator.next();
+      assertNotNull(image.id());
+      assertNotNull(image.imageId());
+      assertNotNull(image.creationTimestamp());
+      assertNotNull(image.configuration());
+      assertNotNull(image.status());
+      assertNotNull(image.diskSizeGb());
+    }
+  }
+
+  @Test
+  public void testListImagesWithSelectedFields() {
+    Page<Image> imagePage =
+        compute.listImages(IMAGE_PROJECT, Compute.ImageListOption.fields(Compute.ImageField.ID));
+    Iterator<Image> imageIterator = imagePage.iterateAll();
+    while (imageIterator.hasNext()) {
+      Image image = imageIterator.next();
+      assertNotNull(image.id());
+      assertNotNull(image.imageId());
+      assertNull(image.creationTimestamp());
+      assertNotNull(image.configuration());
+      assertNull(image.status());
+      assertNull(image.diskSizeGb());
+      assertNull(image.licenses());
+      assertNull(image.deprecationStatus());
+    }
+  }
+
+  @Test
+  public void testListImagesWithFilter() {
+    Page<Image> imagePage = compute.listImages(IMAGE_PROJECT, Compute.ImageListOption.filter(
+        Compute.ImageFilter.equals(Compute.ImageField.ARCHIVE_SIZE_BYTES, 365056004L)));
+    Iterator<Image> imageIterator = imagePage.iterateAll();
+    while (imageIterator.hasNext()) {
```

Comment:
Do you think that we should assert that we got at least one result?  Suppose our filter construction was wrong, we could get no results, and this test would pass.